msrestazure/azure_operation.py

```diff
@@ -108,29 +132,16 @@ def __init__(self, response, outputs):
         self.async_url = None
         self.location_url = None
 
-    def _validate(self, url):
-        """Validate header url
-
-        :param str url: Polling URL extracted from response header.
-        :returns: URL if valid.
-        :raises: ValueError if URL has not scheme or host.
-        """
-        if url is None:
-            return None
-        parsed = urlparse(url)
-        if not parsed.scheme or not parsed.netloc:
-            raise ValueError("Invalid URL header")
-        return url
-
-    def _check_status(self, response):
+    def _raise_if_bad_http_status_and_method(self, response):
         """Check response status code is valid for a Put or Patch
-        reqest. Must be 200, 202, or 204.
+        request. Must be 200, 201, 202, or 204.
 
         :raises: BadStatus if invalid status.
         """
         code = response.status_code
-        if code in [200, 202] or (code == 201 and self.method == 'PUT') or \
-                (code == 204 and self.method in ['DELETE', 'POST']):
+        if code in [200, 202] or \
```

Comment:
Yep, even in Python 2.7 (was backported from 3.0 since they are syntactically unambiguous; same goes for dict and set comprehensions).